src/bios_handler.cpp

```diff
@@ -215,4 +218,42 @@ void IbmBiosHandler::saveFcoToVpd(int64_t i_fcoInBios)
 
     // TODO:  Call Manager API to write keyword data using inventory path.
 }
+
+void IbmBiosHandler::processMemoryMirrorMode()
+{
+    auto l_kwdValueVariant = dbusUtility::readDbusProperty(
+        constants::pimServiceName, constants::systemVpdInvPath,
+        constants::utilInf, constants::kwdAMM);
+
+    if (auto pVal = std::get_if<std::string>(&l_kwdValueVariant))
+    {
+        auto l_ammmValInVpd = *pVal;
+
+        // Check if memory mirror mode value is default in VPD.
+        if (l_ammmValInVpd.at(0) == constants::VALUE_0)
+        {
+            // TODO: Save Ammm to VPD.
+        }
+        else
+        {
+            types::BiosAttributeCurrentValue l_attrValueVariant =
+                readBiosAttribute("hb_memory_mirror_mode");
+
+            if (auto pVal = std::get_if<std::string>(&l_attrValueVariant))
+            {
+                std::string l_ammmInBios = *pVal;
+                (void)l_ammmInBios;
+                // TODO: save to BIOS.
+
+                return;
+            }
+
+            logging::logMessage(
+                "Invalid type recieved for auto memory mirror mode from BIOS.");
```

Comment:
nit: value not type

Reply:
We are doing a type check there for std::string

Reply:
ok